### PR TITLE
snp-sites update to 2.5.1

### PR DESCRIPTION
**Pull request by pvanheus**
FOR CONTRIBUTOR:
* [x] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [x] - License permits unrestricted use (educational + commercial)
* [ ] - This PR adds a new tool or tool collection
* [x] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

### CODE REVIEW

**Comment by pvanheus** [comment]
* Update to snp-sites 2.5.1
* Add options to support all the options of snp-sites
* Add tests for new options
* Update help text

File: tools/snp-sites/snp-sites.xml [hunk 1]
```diff
@@ -1,38 +1,118 @@
...
+         <filter>formats is not None and '-m' in formats and C is False</filter>
+      </data>
+      <data name='output_vcf' format='vcf' label='${tool.name} on ${on_string}: VCF' from_work_dir='output.vcf'>
+         <filter>formats is not None and '-v' in formats and C is False</filter>
+      </data>
+      <data name='output_phylip' format='phylip' label='${tool.name} on ${on_string}: phylip'>
+         <filter>formats is not None and '-p' in formats and C is False</filter>
+      </data>
+      <data name='output_counts' format='tabular' label='${tool.name} on ${on_string}: constant site counts'>
+         <filter>C is True</filter>
```
Comment (bernt-matthias) [inline tools/snp-sites/snp-sites.xml]: Does this work in real life (i.e., when serving .. I guess it does work in the test). I usually test for equality with the truevalue / falsevalue of the parameter.

File: tools/snp-sites/snp-sites.xml [hunk 2]
```diff
@@ -1,38 +1,118 @@
...
+   <macros>
+      <token name="@TOOL_VERSION@">2.5.1</token>
+   </macros>
    <requirements>
-      <requirement type='package' version='2.3.2'>snp-sites</requirement>
+      <requirement type='package' version='@TOOL_VERSION@'>snp-sites</requirement>
    </requirements>
    <version_command>snp-sites -V</version_command>
    <command detect_errors='aggressive'><![CDATA[
+OUTPUTS='$formats' &&
```
Comment (bernt-matthias) [inline tools/snp-sites/snp-sites.xml]: As in the filters I guess you need to check `$fomats is not None`

File: tools/snp-sites/snp-sites.xml [hunk 3]
```diff
@@ -1,38 +1,118 @@
...
-      <requirement type='package' version='2.3.2'>snp-sites</requirement>
+      <requirement type='package' version='@TOOL_VERSION@'>snp-sites</requirement>
    </requirements>
    <version_command>snp-sites -V</version_command>
    <command detect_errors='aggressive'><![CDATA[
+OUTPUTS='$formats' && 
 snp-sites 
--mvp 
+#if str($C) != "-C"
+   \${OUTPUTS//,/ }
```
Comment (bernt-matthias) [inline tools/snp-sites/snp-sites.xml]: Maybe you can solve this also via a cheetah variable, e.g. with

`#set outputs=" ".join($formats.split(","))`

File: tools/snp-sites/snp-sites.xml [hunk 4]
```diff
@@ -1,38 +1,118 @@
...
+      </data>
+      <data name='output_phylip' format='phylip' label='${tool.name} on ${on_string}: phylip'>
+         <filter>formats is not None and '-p' in formats and C is False</filter>
+      </data>
+      <data name='output_counts' format='tabular' label='${tool.name} on ${on_string}: constant site counts'>
+         <filter>C is True</filter>
+      </data>
    </outputs>
    <tests>
       <test>
```
Comment (bernt-matthias) [inline tools/snp-sites/snp-sites.xml]: please add `expect_num_outputs` to `<test>`

File: tools/snp-sites/snp-sites.xml [hunk 5]
```diff
@@ -1,38 +1,118 @@
...
          <param name='input_fasta' value='input.fasta' ftype='fasta' />
+         <param name='formats' value='-m,-p,-v' />
          <output name='output_fasta' value='output.fasta' ftype='fasta' />
          <output name='output_vcf' value='output.vcf' ftype='vcf' />
          <output name='output_phylip' value='output.phylip' ftype='phylip' />
       </test>
+      <test>
+         <param name='input_fasta' value='input.fasta' ftype='fasta' />
+         <param name='formats' value='-m' />
+         <param name='b' value='True' />
```
Comment (bernt-matthias) [inline tools/snp-sites/snp-sites.xml]: `value="-b"`?

File: tools/snp-sites/snp-sites.xml [hunk 6]
```diff
@@ -1,38 +1,118 @@
...
+      && mv 'phylip' '$output_phylip'
+   #end if
+#end if
 ]]></command>
    <inputs>
         <param name='input_fasta' format='fasta' type='data' label='FASTA file' />
+        <param type='select' multiple='true' name='formats' label='Output formats' help='Output format to use if not counting constant sites'>
+         <option value="-m"  selected="true">Multi-FASTA alignment file</option>
+         <option value="-v">VCF</option>
+         <option value="-p">Phylip</option>
```
Comment (bernt-matthias) [inline tools/snp-sites/snp-sites.xml]: Maybe also add `-r     output internal pseudo reference sequence` ... or is this not useful?

File: tools/snp-sites/snp-sites.xml (hunk 1 above)
Comment (bernt-matthias) [reply tools/snp-sites/snp-sites.xml]: An appropriate test will make sure. 

Maybe `<filter>C</filter>` might be sufficient (but maybe `<filter>C is True</filter>` is more obvious).

File: tools/snp-sites/snp-sites.xml (hunk 6 above)
Comment (pvanheus) [reply tools/snp-sites/snp-sites.xml]: I don't think its a useful option. From the code it looks like the pseudo reference sequence is effectively equal to the first sequence in the FASTA in most cases.

**Comment by bgruening** [comment]
@bernt-matthias ready?

**Comment by bgruening** [comment]
@bernt-matthias, please feel free to merge, you have IUC power - use it! :)

**Comment by bernt-matthias** [comment]
@bgruening .. was just waiting for an answer to my last comment: 

> Maybe the -C parameter and the formats parameter should be combined with a conditional, then users can't use the tools in a wrong way.

**Comment by bernt-matthias** [comment]
Also: isn't there a 2 approvals requirement before merging. Maybe I remember wrong.

**Comment by nsoranzo** [comment]
> Also: isn't there a 2 approvals requirement before merging. Maybe I remember wrong.

No, one approval is enough.

**Pull request merged**